courseraresearchexports/containers/client.py

```diff
@@ -0,0 +1,174 @@
+    """
+    logging.info('Starting containers...')
+    docker_client.start(container)
+    # poll logs to see if database is ready to accept connections
+    while POSTGRES_READY_MSG not in docker_client.logs(container, tail=4):
+        logging.debug('Polling container for database connection...')
+        time.sleep(1)
+    logging.info('Container ready.')
+
+
+def stop(container, docker_client):
+    """
+    Stops a docker container
+    """
+    docker_client.stop(container)
+
+
+def remove(container, docker_client):
+    """
+    Remove a stopped container
+    """
+    docker_client.remove_container(container)
+
+
+def create_from_folder(
+        export_data_folder, docker_client, container_name='coursera-exports',
+        database_name='coursera-exports'):
+    """
+    Using a folder containing a Coursera research export, create a docker
+     container with the export data loaded into a data base and start the
+     container
+    :param export_data_folder: folder where export data/scripts is stored
+    :param docker_client:
+    :param container_name:
+    :param database_name:
+    :return container:
+    """
+    try:
+        if not docker_client.images(name=POSTGRES_DOCKER_IMAGE):
+            logging.warn('Downloading image: {}'.format(POSTGRES_DOCKER_IMAGE))
+            docker_client.import_image(image=POSTGRES_DOCKER_IMAGE)
+
+        for existingContainer in docker_client.containers(
+                all=True, filters={'name': container_name}):
+            logging.info('Removing existing containers: {}'.format(
+                container_name))
+            docker_client.remove_container(existingContainer, force=True)
+
+        logging.debug('Creating containers from {folder}'.format(
+            folder=export_data_folder))
+        container = docker_client.create_container(
+            image=POSTGRES_DOCKER_IMAGE,
+            name=container_name,
+            labels=[COURSERA_DOCKER_LABEL],
+            volumes=['/mnt/exportData'],
+            host_config=docker_client.create_host_config(
+                binds=['{}:/mnt/exportData:ro'.format(export_data_folder)],
+                port_bindings={
+                    5432: get_next_available_port(docker_client)
+                })
+        )
+
+        # copy containers initialization script to entrypoint
+        database_setup_script = '''
+            createdb -U {user} {db}
+            cd /mnt/exportData
+            psql -e -U {user} -d {db} -f setup.sql
+            psql -e -U {user} -d {db} -f load.sql
+        '''.format(user='postgres', db=database_name)
+
+        docker_client.put_archive(
+            container,  # using a named argument here causes NullResource error
+            path='/docker-entrypoint-initdb.d/',
+            data=utils.create_tar_archive(
+                database_setup_script, name='init-user-db.sh'))
+
+        logging.info('Created containers with id: {}'.format(container['Id']))
```

Comment:
can we try and use one way of passing in variables into logging? I see we use the `%(VAR_NAME)s` as well. Let's standardize on 1 usage.